string_grouper/test/test_string_grouper.py

```diff
@@ -124,118 +155,138 @@ def test_case_insensitive_build_matches_list(self):
-        with self.assertRaises(Exception):
-            _ = StringGrouper(test_series_1, duplicates_id=test_series_id_2, master_id=test_series_id_1)
+            _ = StringGrouper(test_series_1, duplicates=test_series_2, duplicates_id=bad_test_series_id_2,
+                              master_id=good_test_series_id_1)
 
         # When the input data is ok but the option combinations are invalid:
-        test_series_1 = pd.Series(['foo', 'bar', 'baz'])
-        test_series_id_1 = pd.Series(['A0', 'A1', 'A2'])
-        test_series_2 = pd.Series(['foo', 'bar', 'bop'])
-        test_series_id_2 = pd.Series(['B0', 'B1', 'B2'])
         with self.assertRaises(Exception):
-            _ = StringGrouper(test_series_1, test_series_2, master_id=test_series_id_1)
+            _ = StringGrouper(test_series_1, test_series_2, master_id=good_test_series_id_1)
+        with self.assertRaises(Exception):
+            _ = StringGrouper(test_series_1, test_series_2, duplicates_id=good_test_series_id_2)
         with self.assertRaises(Exception):
-            _ = StringGrouper(test_series_1, test_series_2, duplicates_id=test_series_id_2)
+            _ = StringGrouper(test_series_1, duplicates_id=good_test_series_id_2)
         with self.assertRaises(Exception):
-            _ = StringGrouper(test_series_1, duplicates_id=test_series_id_2)
+            _ = StringGrouper(test_series_1, master_id=good_test_series_id_1, duplicates_id=good_test_series_id_2)
+
+    def test_get_groups_single_df_group_rep(self):
+        """Should return a pd.series object with the same length as the original df. The series object will contain
+        a list of the grouped strings"""
+        simple_example = SimpleExample()
+        customers_df = simple_example.customers_df
+        pd.testing.assert_series_equal(
+            simple_example.expected_result_S,
+            group_similar_strings(
+                customers_df['Customer Name'],
+                group_rep='centroid',
+                min_similarity=0.6
+            )
+        )
         with self.assertRaises(Exception):
-            _ = StringGrouper(test_series_1, master_id=test_series_id_1, duplicates_id=test_series_id_2)
+            _ = group_similar_strings(
+                    customers_df['Customer Name'],
+                    group_rep='nonsense',
+                    min_similarity=0.6
+                )
 
     def test_get_groups_single_df(self):
         """Should return a pd.series object with the same length as the original df. The series object will contain
         a list of the grouped strings"""
         test_series_1 = pd.Series(['foooo', 'bar', 'baz', 'foooob'])
-        sg = StringGrouper(test_series_1)
-        sg = sg.fit()
-        result = sg.get_groups()
-        expected_result = pd.Series(['foooo', 'bar', 'baz', 'foooo'])
-        pd.testing.assert_series_equal(expected_result, result)
+        # sg = StringGrouper(test_series_1)
```

Comment:
Same as above

Reply:
Sure.

Reply:
Not sure if you'd like me to do this for all the other cases where I made similar changes?  Let me know if you do.